packages/styled-components/transforms/tests/fixtures/.transpile-css-prop/output.js

```diff
@@ -1,232 +0,0 @@
-import _styled24 from "styled-components";
-import _styled23 from "styled-components";
-import _styled22 from "styled-components";
-import _styled21 from "styled-components";
-import _styled20 from "styled-components";
-import _styled19 from "styled-components";
-import _styled18 from "styled-components";
-import _styled17 from "styled-components";
-import _styled16 from "styled-components";
-import _styled15 from "styled-components";
-import _styled14 from "styled-components";
-import _styled13 from "styled-components";
-import _styled12 from "styled-components";
-import _styled11 from "styled-components";
-import _styled10 from "styled-components";
-import _styled9 from "styled-components";
-import _styled8 from "styled-components";
-import _styled7 from "styled-components";
-import _styled6 from "styled-components";
-import _styled5 from "styled-components";
-import _styled4 from "styled-components";
-import _styled3 from "styled-components";
-import _styled2 from "styled-components";
-import _styled from "styled-components";
-
-/*
- * Basic fixtures
- */
-const StaticString = p => <_StyledP>A</_StyledP>;
-
-const StaticTemplate = p => <_StyledP2>
-    A
-  </_StyledP2>;
-
-const ObjectProp = p => <_StyledP3>A</_StyledP3>;
-
-const NoChildren = p => <_StyledP4 />;
-
-const CssHelperProp = p => <_StyledP5>
-    A
-  </_StyledP5>;
-/*
- * Dynamic prop
- */
-
-
-const CustomComp = p => <_StyledParagraph>H</_StyledParagraph>;
-
-const DynamicProp = p => <_StyledP6 $_css={props.cssText}>H</_StyledP6>;
-
-const LocalInterpolation = p => <_StyledP7 $_css2={props.bg}>
-    H
-  </_StyledP7>;
-
-const FuncInterpolation = p => <_StyledP8>
-    H
-  </_StyledP8>;
-
-const radius = 10;
-
-const GlobalInterpolation = p => <_StyledP9>
-    H
-  </_StyledP9>;
-
-const LocalCssHelperProp = p => <_StyledP10 $_css3={p.color}>
-    A
-  </_StyledP10>;
-
-const DynamicCssHelperProp = p => <_StyledP11>
-    A
-  </_StyledP11>;
-
-const CustomCompWithDot = p => <_StyledButtonGhost>H</_StyledButtonGhost>;
-
-const NestedCompWithDot = p => <_StyledButtonGhostNew>H</_StyledButtonGhostNew>;
-
-const CustomCompWithDotLowerCase = p => <_StyledButtonGhost2>H</_StyledButtonGhost2>;
-
-const CustomElement = p => <_StyledButtonGhost3>H</_StyledButtonGhost3>;
-/* styled component defined after function it's used in */
-
-
-const EarlyUsageComponent = p => <_StyledThing />;
-
-const Thing3 = styled.div`
-  color: blue;
-`;
-
-var _StyledThing5 = _styled(Thing3)(p => ({
```

Comment:
Isn't this order important?
If it's defined like this, it will be an error if parent component is rendered while defining.

Reply:
Seems like the authors of `styled-components` babel plugin intentionally added tests for this.

Reply:
Yes. I think that's why `styled-components` has related tests.

Reply:
Ordering is determined by https://github.com/styled-components/babel-plugin-styled-components/blob/16e59a84fc5a0de0982cf2c0df58f225808cc783/src/visitors/transpileCssProp.js#L74-L88